### ISSUE TITLE
Messages displayed incorrectly in threads (as compared to element web)

### ISSUE THREAD

**Issue by enygmator** [vector-im/element-android#7857]
### Steps to reproduce

1. Where are you starting? What can you see?
2. What do you click?
3. More steps…

This is pretty difficult to reproduce as this seem to be some kind of intermittently occurring issue and has to do with the speed to which you send messages (some kind of consistency issue?). Steps to reproduce:

1. Open the Element app, go to a chat and write a message, then create a thread using "reply in thread"
2. When I start typing messages in that thread, everything seems fine.
3. Try to **reply to your OWN messages** in that chat, preferably **immediately** after you write the initial message
4. If you check our the other user's thread on element-web, everything appears perfectly
5. Now switch over to the element-android, you _may_ see that the messages show up in a very strange way.

> NOTE: It doesn't happen everytime, especially not if you wait sometime before replying to your own message. Hard to reproduce, but happen enough times to make you lose your mind

The following screenshots are what you will observe:

**web - user 1**:
![web-user-1](https://user-images.githubusercontent.com/36495185/209859629-c8d1c0ad-d252-48b9-aa72-446f3e2dc410.jpg)

**web - user 2**:
![web-user-2](https://user-images.githubusercontent.com/36495185/209859649-5ca3e81c-e377-45cc-99ab-4dd2467bde8a.jpg)

Everything seems good? Now look at **android - user 1**:
![android-user-1](https://user-images.githubusercontent.com/36495185/209859679-7c97519e-6f5e-4c4c-b452-629cf3c1ab60.jpg)

## Side effects

I have also noticed a side effect of this. On element-android, when "editing" a normal message, clicking the "edit" button will put the message in the textbox below, where you can edit it and hit the "tick" button to finish editing. 

1. BUT, when the messages are displayed incorrectly (the way it's shown above), hitting the "edit" button will show an **empty textbox** and whatever you write in there will replace the entire text in the message you are trying to edit. 
2. BUT, then the message seems to be correctly showing "In reply to a thread" on the phone. 
3. BUT, on element-web, it shows up even more weirdly.
4. Then, editing on element-web will make things look normal on the web, but then android looks weird.

You can observe in following screenshots:

**This is how the messages look on android (initially)** (as you can see, they say they are in reply to something, but the UI is deformed and we don't know what).

![android-iinitial](https://user-images.githubusercontent.com/36495185/209861744-fe9bb79e-c433-4e5e-b719-a2b7d0bb3228.jpg)

Now, I'll make an edit on android by hitting the edit button. (the textbox is empty so I type is whatever I want and hit the "tick"). **you can see the result of the edit on android**:

![android-after-edit](https://user-images.githubusercontent.com/36495185/209861836-6b6060fb-7798-457c-8049-7b7d7e55d7d5.jpg)

**Now, I'll show you how the edit looks on element-web**. NOTE that the message below the edited message is how the edited message kind of looked like on element-web, originally.

![web-user-1-after](https://user-images.githubusercontent.com/36495185/209861962-f8c5e856-15fc-439b-9daf-faab2b87bb2c.jpg)

Now, I'll edit the SAME message on element-web:

![web-user-1-after-edit](https://user-images.githubusercontent.com/36495185/209861999-7babe68b-5cc9-4976-9da9-20cb8b093c16.jpg)

After I'm done editing on the web, it looks like this:

![web-user-1-after-edit-after](https://user-images.githubusercontent.com/36495185/209862030-ebebd8a3-de0b-4e9a-90ba-4860502ecf19.jpg)

Now, let's go back and see how the re-edit looks on android:

![android-after-edit-on-web](https://user-images.githubusercontent.com/36495185/209862079-d12f3517-5f3a-444a-9f36-a6d9e8b5671b.jpg)

As you can see, this whole thing seems like a real mess. I also uncovered another bug which I couldn't understand why it was happening. I'll create it later and maybe link it here. It may be related.

So, to summarize, the two issues are:
1. the weird showing of "in reply to thread"
2. the inability to edit properly and everything showing up stangely in UI (thus impacting experience on element-web too, which actually seem fine, if element-android didn't incorrectly "edit" the messages.

Thanks!

#### What did you expect?

I'm sorry, I put the screentshots in "steps to reproduce"

#### What happened instead?

I'm sorry, I put the screentshots in "steps to reproduce"

### Your phone model

Xiaomi Redmi Note 9

### Operating system version

Android 11

### Application version and app store

Element version 1.5.14, olm version 3.2.12 from  playstore

### Homeserver

matrix.org

### Will you send logs?

Yes

### Are you willing to provide a PR?

No

**Comment by enygmator** [comment]
I'm not sure how github allows change of image size

**Comment by enygmator** [comment]
I updated the description to properly describe the issues (bugs) and suggestions.
